fcrepo-kernel-modeshape/src/main/java/org/fcrepo/kernel/modeshape/observer/SimpleObserver.java

```diff
@@ -184,7 +185,7 @@
     @PostConstruct
     public void buildListener() throws RepositoryException {
         LOGGER.debug("Constructing an observer for JCR events...");
-        session = repository.login();
+        session = getJcrSession(repository.login());
         session.getWorkspace().getObservationManager()
                 .addEventListener(this, EVENT_TYPES, "/", true, null, null, false);
         session.save();
```

Comment:
@awoods I think this is kosher-- there is no conceivable way any outside-the-kernel-impl process could ever become aware of this session, and inside the kernel-impl is exactly where we expect to find platform-bound constructs like this.